src/containers/DownloadContainer/DownloadContainer.jsx

```diff
@@ -0,0 +1,120 @@
+import React, { useEffect, useState } from 'react'
+import { FlatList, View } from 'react-native'
+import PropTypes from 'prop-types'
+import * as FileSystem from 'expo-file-system'
+import { Icon, Image, ListItem } from 'react-native-elements'
+import * as VideoThumbnails from 'expo-video-thumbnails'
+import { STACK_SCREENS } from '../CTNavigationContainer/index'
+
+/**
+ * Contains the home screen of the application. Lists starred courses and gives the user the ability
+ * to search for courses. Clicking on a course shows the playlists for it.
+ */
+const DownloadContainer = ({ navigation }) => {
+  const [localVideos, setLocalVideos] = useState([]) // an array of local video names
+
+  const getFileSize = async (fileUri) => {
+    const fileInfo = await FileSystem.getInfoAsync(fileUri)
+    return (fileInfo.size / (1024 * 1024)).toFixed(2)
+  }
+
+  const generateThumbnail = async (videoPath) => {
+    try {
+      const { uri } = await VideoThumbnails.getThumbnailAsync(videoPath, {
+        time: 15000,
+      })
+      return uri
+    } catch (e) {
+      // eslint-disable-next-line no-console
+      console.warn(e)
+      return null
+    }
+  }
+
+  const fetchLocalFiles = async () => {
+    let files = await FileSystem.readDirectoryAsync(FileSystem.documentDirectory)
+    files = files.filter((fileName) => fileName.endsWith('.mp4'))
+    const videoItems = await Promise.all(
+      files.map(async (fileName) => {
+        const localUri = FileSystem.documentDirectory + fileName
+        const thumbnail = await generateThumbnail(localUri)
+        const filesize = await getFileSize(localUri)
+        return {
+          name: fileName.slice(0, -4),
+          thumbnailImg: thumbnail,
+          filesize,
+          video: {
+            video1Path: fileName,
+          },
+        }
+      })
+    )
+    setLocalVideos(videoItems)
+  }
+
+  useEffect(() => {
+    fetchLocalFiles()
+    const interval = setInterval(() => {
+      fetchLocalFiles()
+    }, 5000)
```

Comment:
What's the reason for this periodic fetching

Reply:
Is that for if someone starts a download then goes to the download screen before it finishes?

Reply:
> Is that for if someone starts a download then goes to the download screen before it finishes?
Yes, exactly. This just checks the videos folder every 5 seconds.